app/models/proxy.rb

```diff
@@ -278,15 +278,16 @@ def publish_events
     nil
   end
 
-  DEPLOYMENT_OPTION_CHANGED = ->(record) { record.changed_attributes.key?(:deployment_option) }
-
+  # TODO: remove unused method, there is no such attribute
   def deployment_option_changed?
-    [ self, service ].any?(&DEPLOYMENT_OPTION_CHANGED)
+    [self, service].any? do |record|
+      record.changed_attributes.key?(:deployment_option)
+    end
   end
 
   # We want to autosave when Service#deployment_option changed
   def changed_for_autosave?
-    deployment_option_changed? or super
+    association(:service).loaded? && service.will_save_change_to_attribute?(:deployment_option) or super
```

Comment:
First of all `#deployment_option_changed?` is overriding the deprecated method of the `attribute_changed?` family. So I believe we want to just get rid of it.

I have tried `changed_for_autosave?` method and it returns `false` inside an `after-save` callback as well after a `save` call. So I believe equivalent is `will_save_change..`.

Another thing I checked is that `Proxy` lacks a `:deployment_option` attribute at all. So this seems to be some old code from the distant past when that attribute was being migrated from `Proxy` to `Service`.

That's why I leave only check for `service`.

A small optimization is to check whether `association(:service)` was loaded to avoid a database call in case it was not. If object was not loaded, then it can't have any changes applied so that would be a worthless call.

Reply:
Makes sense!